### ISSUE TITLE
feat: allow sorting of viable snippet optional

### ISSUE THREAD

**Issue by yxliang01** [squaresLab/Darjeeling#261]
At https://github.com/squaresLab/Darjeeling/blob/bba5197719d502e6e651dc16ad81383dc32baaf3/src/darjeeling/transformation/classic/base.py#L146 , the list of potential snippets needs to be generated before the generator can yield new element. This can be very inefficient when the viable snippet space is huge. While determinism is generally preferred, I propose to add a global switch to Darjeeling to toggle whether determinism is required to avoid inefficiencies like this example.

**Comment by ChrisTimperley** [comment]
Good catch. I think that it's definitely desirable to remove the requirement that snippets be sorted at this particular point. With some minor design changes, I think that the end result (i.e., the stream of transformations) can remain deterministic without needing to perform this operation.
